bundles/org.openhab.binding.miio/README.md

```diff
@@ -6116,4 +6131,21 @@ String customScene "Set Scene" (G_light) {channel="miio:basic:light:customScene"
 Number nightlightBrightness "Nightlight Brightness" (G_light) {channel="miio:basic:light:nightlightBrightness"}
 ```
 
+### Zhimi Heater (zhimi.heater.za1) item file lines
+
+note: Autogenerated example. Replace the id (heater) in the channel with your own. Replace `basic` with `generic` in the thing UID depending on how your thing was discovered.
+
+```java
+Group G_heater "Zhimi Heater" <status>
+Switch power "Power" (G_heater) {channel="miio:basic:heater:power"}
+Number target_temperature "Target Temperature" (G_heater) {channel="miio:basic:heater:target_temperature"}
+Number brightness "Brightness" (G_heater) {channel="miio:basic:heater:brightness"}
+Switch buzzer "Buzzer Status" (G_heater) {channel="miio:basic:heater:buzzer"}
+Number relative_humidity "Relative Humidity" (G_heater) {channel="miio:basic:heater:relative_humidity"}
+Switch childlock "Child Lock" (G_heater) {channel="miio:basic:heater:childlock"}
+Switch HWSwitch "HW Switch" (G_heater) {channel="miio:basic:heater:HWSwitch"}
+Number temperature "Temperature" (G_heater) {channel="miio:basic:heater:temperature"}
```

Comment:
Can you use units here?
```suggestion
Number:Temperature temperature "Temperature" (G_heater) {channel="miio:basic:heater:temperature"}
```

Reply:
> as I create all channels dynamically, is it enough to just change the datatype to Number:Temperature when the channel is created like: Channel newChannel = ChannelBuilder.create(channelUID, datatype). .withLabel(friendlyName).build();

Yes, that is all that would be necessary. At that point you just need to use QuantityType instead of DecimalType for those relevant channels. and also make sure to use `QuantityType.toUnit()` when processing the state from `handleCommand`. 

QuantityType allows users to use different units for their UI. For example a binding may report temperature in C but the user wants it displayed as F.

You don't need to make the transition just yet, but please plan for it.

Reply:
yes, made placeholder #8756 for it.
Temperature updates are fine, but still need test  how to conveniently go from a unit string to a Unit 
e.g. from "Celcius" to  Units.CELSIUS in a generic way